Maze.cpp

```diff
@@ -114,7 +117,7 @@ Path Maze::findPath(CellCoordinate start, CellCoordinate end, Direction facing )
 			}
 
 			// This path is the best so far.
-			adjacentNode->previous = *currentNodeItr;
+			(*currentNodeItr)->next = adjacentNode;
```

Comment:
I think the we should still perform the pathfinding from end->start so that all the paths will point to the goal. Since our end position changes less frequently than the start position, we can get the most benefit. In the image above the pathfinding was performed start->end resulting in the neighboring cells pointing back to the start which is less useful.